src/main/groovy/org/gradle/android/workarounds/RoomSchemaLocationWorkaround.groovy

```diff
@@ -104,12 +112,32 @@ class RoomSchemaLocationWorkaround implements Workaround {
 
                 project.gradle.taskGraph.beforeTask {
                     if (it == task) {
-                        configureSchemaLocationOutputs(task)
+                        def relativePath = getConfiguredSchemaLocationForKaptWithoutKotlinc(task)
+                        if (relativePath) {
+                            configureSchemaLocationOutputs(task, relativePath)
+                        }
+                    }
+                }
+
+                doFirst {
+                    setRoomSchemaLocationToTaskSpecificDirForKaptWithoutKotlinc(task)
+                }
+            }
+
+            project.tasks.withType(kaptWithKotlincTaskClass) { Task task ->
```

Comment:
can we lazily configure the task? or would this break the workarounds?
```suggestion
            project.tasks.withType(kaptWithKotlincTaskClass).configureEach { Task task ->
```

Reply:
Good catch!  This was just an oversight on my part.  Fixed.